File: 551_Student_Attendance_Record_I.java
/*
https://leetcode.com/problems/student-attendance-record-i/
You are given a string representing an attendance record for a student. The record only contains the following
three characters:
'A' : Absent.
'L' : Late.
'P' : Present.
A student could be rewarded if his attendance record doesn't contain more than one 'A' (absent) or more than two
continuous 'L' (late). Return whether the student could be rewarded according to his attendance record.

Regular expression matching.
Time - O(n)
Space - O(1)
*/

class Solution {
    public boolean checkRecord(String s) {
        // .*A.*A.* means any char zero or more times before, between and after two As
        // .*LLL.* means any char before and after "LLL"
        return !s.matches(".*A.*A.*|.*LLL.*");

    }
}

File: 551_Student_Attendance_Record_I.py
_author_ = 'jake'
_project_ = 'leetcode'

# https://leetcode.com/problems/student-attendance-record-i/
# You are given a string representing an attendance record for a student.
# The record only contains the following three characters:
# 'A' : Absent.
# 'L' : Late.
# 'P' : Present.
# A student could be rewarded if his attendance record doesn't contain more than one 'A' (absent) or more than
# two continuous 'L' (late).
# You need to return whether the student could be rewarded according to his attendance record.

# Find the count of 'A' and whether 'LLL' appears in s.
# Time - O(n)
# Space - O(1)

class Solution(object):
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        return s.count("A") < 2 and "LLL" not in s